### PR TITLE
Update packet header sizes

### PR DESCRIPTION
**Pull request by robgjansen**
- Updates TCP+IP header sum to be 20+20=40 instead of 52
- Added config for TCP MSS = MTU(1500) - IP(20) - TCP(20) = 1460 bytes
- Remove all uses of the ethernet header size, so we stop pretending that we simulate layer 2
- Refactored packet header and payload size accessors

### CODE REVIEW

**Comment by robgjansen** [comment]
Performance benchmark is here:
https://github.com/shadow/benchmark-results/tree/master/tor/2022-07-26-T21-01-40

The perf plot diffs seem reasonable to me.

File: src/main/host/descriptor/tcp.c
Comment (RWails) [inline src/main/host/descriptor/tcp.c]: Consider using the `CONFIG_TCP_MAX_SEGMENT_SIZE` constant defined in `src/main/core/support/definitions.h`.

File: src/main/host/descriptor/tcp.c
Comment (RWails) [inline src/main/host/descriptor/tcp.c]: Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.

File: src/main/host/descriptor/tcp.c
Comment (RWails) [inline src/main/host/descriptor/tcp.c]: Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.

File: src/main/host/descriptor/tcp.c
Comment (RWails) [inline src/main/host/descriptor/tcp.c]: Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.

File: src/main/host/descriptor/tcp.c
Comment (RWails) [inline src/main/host/descriptor/tcp.c]: Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.

File: src/main/host/network_interface.c
Comment (RWails) [inline src/main/host/network_interface.c:455]: In a couple places throughout the code, you're using the snippet `packet_getPayloadLength(packet) + packet_getHeaderSize(packet)`.

You could de-duplicate by defining a convenience function in `src/main/routing/packet.h`:

```C
/*                                                                                                                                                                                                                                                                                                                                                                                                              
 * Returns the packet's total length in bytes, including both header and payload fields.                                                                                                                                                                                                                                                                                                                        
 */                                                                                                                                                                                                                                                                                                                                                                                                             
static inline gsize packet_getTotalLength(const Packet *packet) {                                                                                                                                                                                                                                                                                                                                               
    return (gsize) packet_getPayloadLength(packet) + (gsize) packet_getHeaderSize(packet);                                                                                                                                                                                                                                                                                                                      
}   
```

File: src/main/host/network_interface.c
Comment (RWails) [inline src/main/host/network_interface.c:620]: See `packet_getTotalLength()` suggestion.

File: src/main/routing/router_queue_codel.c
Comment (RWails) [inline src/main/routing/router_queue_codel.c:110]: Ah, OK, you did define a convenience function here. I think a better place for this to live is `src/main/routing/packet.h`, so that other callers can use this too.

**Review by RWails — APPROVED** [review]
Thanks Rob! Looks good.

Only small changes suggested. I'll leave it to your discretion if you want to make the changes before merging.

File: src/main/host/network_interface.c
Comment (robgjansen) [reply src/main/host/network_interface.c]: OK I refactored a `packet_getTotalSize` function as suggested.

**Pull request merged**
